src/app/hearing-part/effects/hearing-part.effects.ts

```diff
@@ -27,9 +27,9 @@ export class HearingPartEffects {
         ofType<AssignToSession>(HearingPartActionTypes.AssignToSession),
         mergeMap(action =>
             this.hearingPartService.assignToSession(action.payload).pipe(
-                mergeMap(data => [new AssignComplete(data.entities.hearingParts),
+                mergeMap(data => [new AssignComplete({id: action.payload.hearingPartId, session: action.payload.sessionId}),
                     new Notify(HEARING_PART_ASSIGN_SUCCESS)]),
-                catchError((err: HttpErrorResponse) => of(new AssignFailed(err.error)))
+                catchError((err) => of(new notificationActions.OpenDialog(HEARING_PART_DIALOGS[err])))
```

Comment:
what happens when err-or is not in the HEARING_PART_DIALOGS array?

Reply:
It will probably return undefined. Would you propose to put HEARING_PART_DIALOGS in a class, create a proper getter that throws exception on not existant element?